app/models/user_role.rb

```diff
@@ -139,10 +139,32 @@ def discourse_user_group
   end
 
   def can_user_read?(user)
-    # A user can view a role if:
-    # 1. the role belongs to a non-hidden group, or
-    # 2. the user has read-access to that group.
-    !group.is_hidden || user&.has_permission?(:can_read_groups, group.id)
+    return true unless group.is_hidden
+    return false if user.nil?
+    if group.banned_competitors?
+      if is_active? # The actively banned competitors can be viewed by following set of people:
+        (
+          user.wdc_team? || # 1. WDC
+          user.ethics_committee? || # 2. WEC
+          user.board_member? || # 3. Board
+          user.staff_delegate? || # 4. Staff Delegates
+          user.weat_team? || # 5. WEAT
+          user.results_team? || # 6. WRT
+          user.admin? # 7. Admin
```

Comment:
This logic can be moved to user.rb, but I felt having this here itself will be better, mainly because this won't be used anywhere else. Also this makes the function more easy to understand.
Same with the else part.